File: src/main/kotlin/Credit.java
import jdk.jshell.Snippet;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public class Credit extends Customer {
    private UUID creditCode;
    private BigDecimal creditValue;
    private LocalDate dayFirstInstallment;
    private int numberOffInstallments;
    private interface Status status;

    private Customer customer;
    private Long id;
}

File: src/main/kotlin/entity/Credit.kt
package entity

import enumeration.Status
import java.math.BigDecimal
import java.time.LocalDate
import java.util.UUID

data class Credit (
    val creditCode: UUID = UUID.randomUUID(),
    val creditValue: BigDecimal= BigDecimal.ZERO,
    val dayFirstInstallment : LocalDate,
    val numberOfInstaments: Int=0,
    val status: Status = Status.IN_PROGRESS,
    val customer: Customer? = null,
    val id: Long?=null
)